From bf769acfd7e63f083c40781c78023ae71fcea21f Mon Sep 17 00:00:00 2001
From: Yann Gautier <yann.gautier@foss.st.com>
Date: Fri, 20 Oct 2023 17:10:04 +0200
Subject: [PATCH 381/954] refactor(stm32mp2): update IMAGE_IDs

Try to align as much as possible the IMAGE_ID definitions with STM32MP1.
Decrease DDR_FW_ID and MAX_NUMBER_IDS, as ID 25 did not seem to be used.
And correct a typo MP1 -> MP2.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git 08c214f890bd5fca19d068727c9616c082a6cfac refactor(stm32mp2): update IMAGE_IDs]
Change-Id: Ic952a14d0d15ba185cfce8dab1d32610133e63e0
Signed-off-by: Yann Gautier <yann.gautier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/335168
Reviewed-by: Nicolas LE BAYON <nicolas.le.bayon@st.com>
---
 plat/st/stm32mp2/include/plat_tbbr_img_def.h | 30 +++++++++++++-------
 1 file changed, 20 insertions(+), 10 deletions(-)

diff --git a/plat/st/stm32mp2/include/plat_tbbr_img_def.h b/plat/st/stm32mp2/include/plat_tbbr_img_def.h
index 8d319fa430..e786dd2b12 100644
--- a/plat/st/stm32mp2/include/plat_tbbr_img_def.h
+++ b/plat/st/stm32mp2/include/plat_tbbr_img_def.h
@@ -18,26 +18,36 @@
 #undef NT_FW_CONFIG_ID
 #undef SOC_FW_CONFIG_ID
 #undef TB_FW_CONFIG_ID
-
-/* Define the STM32MP1 used ID */
-#define BKUP_FWU_METADATA_IMAGE_ID	U(1)
-#define FWU_METADATA_IMAGE_ID		U(2)
-#define FW_CONFIG_ID			U(8)
-#define ENC_IMAGE_ID			U(12)
-#define GPT_IMAGE_ID			U(16)
+#undef HW_CONFIG_ID
+#undef TRUSTED_BOOT_FW_CERT_ID
+#undef SOC_FW_CONTENT_CERT_ID
+#undef BL32_EXTRA1_IMAGE_ID
+#undef TOS_FW_CONFIG_ID
+
+/* Define the STM32MP2 used ID */
+#define FW_CONFIG_ID			U(1)
+#define HW_CONFIG_ID			U(2)
+#define ENC_IMAGE_ID			U(6)
+#define BL32_EXTRA1_IMAGE_ID		U(8)
+#define FWU_METADATA_IMAGE_ID		U(12)
+#define BKUP_FWU_METADATA_IMAGE_ID	U(13)
+#define TOS_FW_CONFIG_ID		U(16)
 #define NT_FW_CONFIG_ID			U(18)
 #define SOC_FW_CONFIG_ID		U(19)
 #define TB_FW_CONFIG_ID			U(20)
+#define TRUSTED_BOOT_FW_CERT_ID		U(21)
+#define SOC_FW_CONTENT_CERT_ID		U(23)
 #define STM32MP_CONFIG_CERT_ID		U(24)
+#define GPT_IMAGE_ID			U(25)
 
 #if STM32MP_DDR_FIP_IO_STORAGE
-#define DDR_FW_ID			U(27)
+#define DDR_FW_ID			U(26)
 /* Increase the MAX_NUMBER_IDS to match the authentication pool required */
-#define MAX_NUMBER_IDS			U(28)
+#define MAX_NUMBER_IDS			U(27)
 
 #else
 /* Increase the MAX_NUMBER_IDS to match the authentication pool required */
-#define MAX_NUMBER_IDS			U(27)
+#define MAX_NUMBER_IDS			U(26)
 
 #endif
 
-- 
2.53.0

